controllers/networkinterface_controller.go

```diff
@@ -1166,6 +1166,34 @@ func (r *NetworkInterfaceReconciler) reconcileFirewallRules(ctx context.Context,
 	return nil
 }
 
+func (r *NetworkInterfaceReconciler) getInterfaceMeteringParams(nic *metalnetv1alpha1.NetworkInterface) (*dpdk.MeteringParams, error) {
+
+	meterParams := &dpdk.MeteringParams{
+		TotalRate:  0,
+		PublicRate: 0,
+	}
+
+	if nic.Spec.MeteringRate == nil {
+		return meterParams, nil
+	}
+
+	if nic.Spec.MeteringRate.TotalRate != nil && nic.Spec.MeteringRate.PublicRate != nil {
+		if *nic.Spec.MeteringRate.TotalRate < *nic.Spec.MeteringRate.PublicRate {
```

Comment:
When we reach this point of reconciliation, it is a bit too late to verify the Spec, as we already claimed an interface for example and it will stay blocked, till this interface gets its Spec corrected.
We already have a function which is called ` isValidIPConfiguration()` which can be renamed to` isValidConfiguration()` and can also check the saneness of the MeteringRate, if set and set the interface to "error" state in the early stage of the reconcilation.

Reply:
good point. revised accordingly.